src/data_race.rs

```diff
@@ -132,6 +116,12 @@ struct ThreadClockSet {
     /// The last timestamp of happens-before relations that
     /// have been released by this thread by a fence.
     fence_release: VClock,
+
+    pub fence_seqcst: VClock,
+
+    pub write_seqcst: VClock,
+
+    pub read_seqcst: VClock,
```

Comment:
Could you add doc comments for these?

Reply:
I added some comments detailing what they do, although it's really difficult to explain why they are needed. Frankly I don't have an intuitive understanding of these bits in the paper either